main.tf

```diff
@@ -1,116 +1,128 @@
-  instance_type = "${var.instance_type}"
-  key_name = "${var.ssh_key_name}"
-  security_groups = ["${aws_security_group.sg.name}"]
+  ami                         = coalesce(var.ami, data.aws_ami.ami.image_id)
+  instance_type               = var.instance_type
+  key_name                    = var.ssh_key_name
+  security_groups             = [aws_security_group.sg.name]
   associate_public_ip_address = true
+  iam_instance_profile        = aws_iam_instance_profile.profile.name
 
-  tags {
-    Name = "${var.project_name}"
+  tags = {
+    Name = var.project_name
   }
 
   root_block_device {
-    volume_size = "${var.volume_size}"
+    volume_size = var.volume_size
   }
 
   connection {
-    user = "ubuntu"
-    private_key = "${file("${var.ssh_private_key}")}"
+    host        = self.public_ip
+    user        = "ubuntu"
+    private_key = file(var.ssh_private_key)
   }
 
   provisioner "file" {
-    content = "${file("docker-compose.yml")}"
-    destination = "~/docker-compose.yml"
+    source      = coalesce(var.cloudwatch_config_json, "${path.module}/cwagentconfig.json")
+    destination = "~/cwagentconfig"
   }
 
   provisioner "file" {
-    content = "${var.init_script}"
-    destination = "~/init.sh"
-  }
-
-  provisioner "remote-exec" {
-    inline = [
-      "mkdir -p ~/config"
-    ]
+    content     = file("docker-compose.yml")
+    destination = "~/docker-compose.yml"
   }
 
   provisioner "file" {
-    source = "${path.cwd}/config/"
-    destination = "~/config/"
+    content     = var.init_script
+    destination = "~/init.sh"
   }
 
   provisioner "remote-exec" {
-    inline = [
-      "sudo apt-get install -y apt-transport-https ca-certificates",
-      "sudo apt-key adv --keyserver hkp://ha.pool.sks-keyservers.net:80 --recv-keys 58118E89F3A912897C070ADBF76221572C52609D",
-      "echo \"deb https://apt.dockerproject.org/repo ubuntu-xenial main\" | sudo tee /etc/apt/sources.list.d/docker.list",
-      "sudo apt-get update",
-      "sudo DEBIAN_FRONTEND=noninteractive apt-get upgrade -y -o DPkg::options::=\"--force-confdef\" -o DPkg::options::=\"--force-confold\"",
-      "sudo apt-get install -y docker-engine",
-      "sudo service docker start",
-      "sudo usermod -aG docker $USER",
-      "sudo curl -L https://github.com/docker/compose/releases/download/1.21.2/docker-compose-$(uname -s)-$(uname -m) -o /usr/local/bin/docker-compose",
-      "sudo chmod +x /usr/local/bin/docker-compose"
-    ]
+    script = "${path.module}/docker-install.sh"
   }
 
   provisioner "remote-exec" {
     inline = [
+      "sudo mv ~/cwagentconfig /etc/cwagentconfig",
+      "sudo docker run -d -v /etc/cwagentconfig:/etc/cwagentconfig amazon/cloudwatch-agent",
       "docker login quay.io -u ${var.quay_username} -p ${var.quay_password}",
       "chmod +x ./init.sh",
-      "docker run -itd --restart always quay.io/buildo/bellosguardo:${var.bellosguardo_target}",
-      "./init.sh"
+      "./init.sh",
+      "sudo docker-compose up -d"
```

Comment:
True, since we added docker group to current user.

Will remove entirely by the way, for the reasons [above](https://github.com/buildo/terraform-aws-dockercomposehost/pull/10#discussion_r483583648)